core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -312,7 +312,15 @@ class LogManager(logDirs: Seq[File],
       val logDirAbsolutePath = dir.getAbsolutePath
       var hadCleanShutdown: Boolean = false
       try {
-        val pool = Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir)
+        val pool = Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir, new ThreadFactory {
+          private val factory = Executors.defaultThreadFactory()
+          private val threadNumber = new AtomicInteger(1)
+          override def newThread(r: Runnable): Thread = {
+            val thread = factory.newThread(r)
+            thread.setName(s"log-recovery(${dir.getAbsolutePath}, ${threadNumber.getAndIncrement()})")
```

Comment:
we can replace `dir.getAbsolutePath` with local variable `logDirAbsolutePath`

Reply:
nit: Do you think it is better to use dash (-) to concat path and threadNumber?
s"log-recovery(${dir.getAbsolutePath}**-**${threadNumber.getAndIncrement()})"

Reply:
`logDirAbsolutePath`: Thanks!

better to use dash (-): I don't have a very strong opinion, except dash is more commonly used in a path name than comma, so if the log dir names were like `data`, `data-2`, then it's a bit confusing because the threads could be named like `data-1`, `data-2`, `data-2-1`, `data-2-2` (assuming >=2 threads per log dir), so people might see some problem associated with the thread `data-2` and think that was about the `data-2` directory, when it's really the 2nd thread processing the `data` directory.